android/modules/calendar/src/java/ti/modules/titanium/calendar/RecurrenceRuleProxy.java

```diff
@@ -0,0 +1,376 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2017 by Axway, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+package ti.modules.titanium.calendar;
+
+import org.appcelerator.kroll.KrollDict;
+import org.appcelerator.kroll.KrollProxy;
+import org.appcelerator.kroll.annotations.Kroll;
+import org.appcelerator.kroll.common.Log;
+import org.appcelerator.titanium.TiApplication;
+import org.appcelerator.titanium.TiC;
+import org.appcelerator.titanium.util.TiConvert;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Kroll.proxy
+public class RecurrenceRuleProxy extends KrollProxy {
+
+	private static final String TAG = "RecurrenceRule";
+
+	// Keys for daysOfTheWeek dictionary.
+	private final String dayOfWeekKey = "daysOfWeek";
+	private final String weekNumberKey = "week";
+
+	// Keys for end dictionary.
+	private final String until = "endDate";
+	private final String count = "occurrenceCount";
+
+	// Field with the Date of the event begin to provide parity with iOS.
+	// Some information exposed can't be get solely from the recurrence rule.
+	private Date eventBegin;
+	private String rRule;
+
+	//region values that can be set from a Creation Dictionary
+	TiRecurrenceFrequencyType frequency = TiRecurrenceFrequencyType.DAILY;
+	private Integer interval = 1;
+	private int[] daysOfTheMonth = new int[]{};
+	private int[] daysOfTheYear = new int[]{};
+	private int[] monthsOfTheYear = new int[]{};
+	private int[] weeksOfTheYear = new int[]{};
+	private String calendarID;
+	private KrollDict endDictionary = new KrollDict();
+	private KrollDict[] daysOfTheWeek = new KrollDict[]{};
+	//endregion
+
+	// enum for matching Titanium frequency constants with Android recurrence rule words
+	public enum TiRecurrenceFrequencyType
```

Comment:
Let's make this enum `private` instead of `public` since it is only used be the parent class. This gives us the freedom to make major changes to it without fearing that an external 3rd party module might be making use of it without our knowing in the future (avoids unintended breaking changes).

Alternatively, I'm okay with this enum being `public` if pulled out into its own separate Java file, but once done, we should avoid breaking changes. (Personally, I would do this. I think the design of it is fine.)